generator-annotations/src/main/java/io/fabric8/crd/generator/annotation/AdditionalPrinterColumn.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines an additional printer column. Must be placed at the root of the
+ * custom resource.
+ */
+@Repeatable(AdditionalPrinterColumns.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AdditionalPrinterColumn {
+
+  //https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#type
+  enum Type {
```

Comment:
Might be useful to move this to the top-level in case we want to reuse this in a different context in the future?

Reply:
Yes it could be useful (in rare cases), e.g. as a final check on the type if `@PrinterColumn` is used and the type is derived. But we have to be careful with naming. This enum represents types, which are only allowed in printer columns. The name should make this very clear, like `PrinterColumnType`.

Reply:
Ah, missed that it was restricted to printer columns… (trying to follow too many things at once). Then it makes sense to keep it as an inner type.